### PR TITLE
Disable problematic task-node output by default

### PR DESCRIPTION
**Pull request by jgfouca**
This code doesn't compile within the unit testing framework
for mct (cime/scripts/fortran_unit_testing/run_tests.py).

Test suite: scripts_regression_tests N_TestUnitTest
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes [CIME Github issue #]

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review: @worleyph

### CODE REVIEW

**Comment by jgfouca** [comment]
@worleyph , it wouldn't let me assign you as a reviewer, but I'd like you to sign off on this.

**Comment by worleyph** [comment]
@jgfouca , why does it fail? Don't know why a TIMING ifdef makes sense for enabling/disabling it.

**Comment by worleyph** [comment]
Is the problem solely with the calls to t_startf/t_stopf?

**Comment by jedwards4b** [comment]
@worleyph Because the unit tests don't include the perf_mod or gptl and don't know about the new shr_taskmap_write file you added.

**Comment by worleyph** [comment]
So, I don't know what the goal is if these are not being tested, but if we always build the model with TIMING enabled, this is fine by me. @jgfouca , you might add a comment within the ifdef describing why these are ifdef'ed.

**Comment by rljacob** [comment]
What does the unit testing framework test within seq_comm_mct?

**Comment by jedwards4b** [comment]
@worleyph the unit tests aren't there to test the timing code, however if you would like to add that to the unit test framework that's another approach that would work.

**Comment by jedwards4b** [comment]
@rljacob I'm not sure but I think that seq_comm_mct is not tested but needed for other code that is.

**Comment by worleyph** [comment]
@jedwards4b , thanks. I'm asking questions without any understanding of what is going on here. If TIMING is excluded in the testing, this makes sense. The shr_taskmap_write call is not timing, but if irrelevant to what you are trying to test, then including it with the TIMING exclusion is fine with me.

**Comment by jgfouca** [comment]
@jedwards4b , just grepping around, it's pretty clear that pio build uses -DTIMING, but it's not as clear to me that mct does.

**Comment by jedwards4b** [comment]
I am confused about the status of testing this PR - @jgfouca can you clarify.

**Comment by jgfouca** [comment]
@jedwards4b , I tested N_TestUnitTest

**Comment by jgfouca** [comment]
I did not test if this code is enabled during a normal run, I'll do that.

**Comment by worleyph** [comment]
@jgfouca , output is near the top of the e3sm.log (pretty much first model output after the PIO options description, though some systems output system environment variable settings first). An F case would be sufficient as well.

Look for something that looks like

48 pes participating in computation of coupled model
--------------------------------------------------------------
GLOBAL communicator : 2 nodes, 48 MPI tasks
COMMUNICATOR NODE # [NODE NAME] : (# OF MPI TASKS) TASK # LIST
GLOBAL NODE 0 [ b634 ] : ( 36 MPI TASKS ) ...
GLOBAL NODE 1 [ b635 ] : ( 12 MPI TASKS ) ...

**Comment by jgfouca** [comment]
@worleyph , i'm testing within cime standalone, so I only have the basic compsets available (A, X, etc).

**Comment by jedwards4b** [comment]
I don't see any reason you wouldn't see it with an A case.

**Comment by jgfouca** [comment]
OK, I've pushed a commit that appears to work. N_TestUnitTest doesn't have a build failure and I've confirmed that a simple test case still has the task-node map output.

@jedwards4b , please re-review. I had to touch the main Makefile. The unit tests use CMake so they don't see this flag.

**Comment by jedwards4b** [comment]
I don't understand why you did this instead of ifdef timing?

**Comment by jgfouca** [comment]
@jedwards4b , I checked, mct doesn't use TIMING

**Comment by jgfouca** [comment]
And I wasn't sure I wanted to turn on TIMING for everything because that could impact pio.

**Comment by jedwards4b** [comment]
TIMING is on by default but is currently only used as a macro in PIO.   I guess that the unit tests didn't incorporate any perf_mod until now.   It would work to use ifdef TIMING here but you would then need to add it to the Makefile for the csm_share build.   I still think that may be better than creating a new ifdef term in the Makefile.

**Comment by jgfouca** [comment]
@jedwards4b , OK , let me try that.

**Comment by jgfouca** [comment]
@jedwards4b , done. Thanks for the help, I was a bit lost in our build systems.

File: src/drivers/mct/shr/seq_comm_mct.F90 [hunk 1]
```diff
@@ -20,8 +20,10 @@ module seq_comm_mct
   use shr_sys_mod    , only : shr_sys_abort, shr_sys_flush
   use shr_mpi_mod    , only : shr_mpi_chkerr, shr_mpi_bcast, shr_mpi_max
   use shr_file_mod   , only : shr_file_getUnit, shr_file_freeUnit
+#ifdef TIMING
```
Comment (rljacob) [inline src/drivers/mct/shr/seq_comm_mct.F90]: Add a comment.  This is for unit testing.  TIMING is on by defaut when compiling cime normally.  Unit tests for Fortran don't need gptl AND have TIMING undefined so add this ifdef for unit tests.

**Pull request merged**